pkg/certificate/x509.go

```diff
@@ -0,0 +1,147 @@
+package certificate
+
+import (
+	"crypto"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"math/big"
+	"os"
+	"path/filepath"
+	"runtime"
+	"time"
+)
+
+const (
+	rootName    = "rootCA.pem"
+	rootKeyName = "rootCA-key.pem"
+)
+
+func GenerateSignedCertificate(serverName string, priv *rsa.PrivateKey) ([]byte, *rsa.PrivateKey, error) {
+	if len(serverName) == 0 {
+		return nil, nil, errors.New("while generating certificate: server name is empty")
+	}
+
+	caPath, err := mkcertCARootPath()
+	if err != nil {
+		return nil, nil, fmt.Errorf("while generating certificate: %w", err)
+	}
+
+	caCert, err := LoadCertificate(caPath + "/" + rootName)
+	if err != nil {
+		return nil, nil, fmt.Errorf("while loading CA certificate: %w", err)
+	}
+
+	caPrivateKey, err := LoadPrivateKey(caPath + "/" + rootKeyName)
+	if err != nil {
+		return nil, nil, fmt.Errorf("while loading CA key: %w", err)
+	}
+
+	//nolint:exhaustruct
+	tpl := &x509.Certificate{
```

Comment:
I think this variable should be renamed because tpl is not straightforward to understand

Reply:
changed to template